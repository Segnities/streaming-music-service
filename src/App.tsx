

function App() {

  return (
    <div className="bg-blue-700 min-h-screen">
      
    </div>
  );
}

export default App
