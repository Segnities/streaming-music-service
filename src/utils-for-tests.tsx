
import {render} from "@testing-library/react";
import {combineReducers, configureStore} from "@reduxjs/toolkit";
import {Provider, useDispatch} from "react-redux";

import playerSlice from "./store/reducers/player";
import firebaseUsersSlice, {setFirebaseUsers} from "./store/reducers/firebaseUsers";
import {shazamCoreApiV1, shazamCoreApiV2} from "./API/shazamCore";

import {getDefaultMiddleware} from "@reduxjs/toolkit";

import {CurriedGetDefaultMiddleware} from "@reduxjs/toolkit/dist/getDefaultMiddleware";
import {AuthContext} from "./context";
import {useEffect, useState} from "react";
import {getAuth, onAuthStateChanged, User} from "firebase/auth";
import {firebaseApp} from "./firebase/firebaseConfig";
import {getUsers} from "./utils/getUsers";

const rootReducer = combineReducers({
    player: playerSlice.reducer,
    firebaseUsers: firebaseUsersSlice.reducer,
    [shazamCoreApiV1.reducerPath]: shazamCoreApiV1.reducer,
    [shazamCoreApiV2.reducerPath]: shazamCoreApiV2.reducer,
});

export function renderWithProviders(
    ui: JSX.Element,
    {
        preloadedState = {},
        store = configureStore({
            reducer: rootReducer,
            middleware: (getDefaultMiddleware: CurriedGetDefaultMiddleware) =>
                getDefaultMiddleware().concat(shazamCoreApiV1.middleware).concat(shazamCoreApiV2.middleware),
            preloadedState
        }),
        ...renderOptions
    } = {}
) {


    function Wrapper({children}) {
        return <Provider store={store}>
                {children}
        </Provider>
    }

    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}
