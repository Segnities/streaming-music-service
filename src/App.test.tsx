
import {describe, expect, test} from "vitest";
import {screen} from "@testing-library/react";

import {renderWithProviders} from "./utils-for-tests";
import App from "./App";
describe('Simple tests', ()=> {
    test('1 + 1 = 2', ()=> {
        expect(1 + 1).toBe(2);
    });
    test('Test after creating tests util', ()=> {
        renderWithProviders(<App/>);
        expect(screen.getByTestId('app-loader')).toBeInTheDocument();
    })
})