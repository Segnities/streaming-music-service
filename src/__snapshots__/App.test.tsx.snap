// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test app component > Loader before while data loading 1`] = `
<div
  class="w-full h-screen flex flex-1 flex-col bg-gradient-to-br from-black to-[#121286]"
  data-testid="app-loader"
>
  <div
    class="w-full flex justify-center items-center flex-col"
  >
    <img
      class="w-32 h-32 object-contain"
      src="/src/components/UI/Loader/assets/img/loader.svg"
    />
    <h1
      class="font-bold text-white text-2xl mt-2"
    >
      Loading...
    </h1>
  </div>
</div>
`;

exports[`Test app component > Loader while data is loading 1`] = `
<div
  class="w-full h-screen flex flex-1 flex-col bg-gradient-to-br from-black to-[#121286]"
  data-testid="app-loader"
>
  <div
    class="w-full flex justify-center items-center flex-col"
  >
    <img
      class="w-32 h-32 object-contain"
      src="/src/components/UI/Loader/assets/img/loader.svg"
    />
    <h1
      class="font-bold text-white text-2xl mt-2"
    >
      Loading...
    </h1>
  </div>
</div>
`;
